src-electron/db/query-feature.js

```diff
@@ -24,53 +24,612 @@ const dbApi = require('./db-api.js')
+  arg.push(...deviceTypeRefs)
+  arg.push(endpointTypeRef)
   let features = await dbApi.dbAll(
     db,
     `
     SELECT
-        d.DESCRIPTION AS DEVICE_TYPE_NAME,
-        dc.CLUSTER_NAME,
-        dc.INCLUDE_SERVER,
-        dc.INCLUDE_CLIENT,
-        df.DEVICE_TYPE_CLUSTER_CONFORMANCE,
-        f.FEATURE_ID,
-        f.NAME AS FEATURE_NAME,
-        f.CODE,
-        f.BIT,
-        f.DEFAULT_VALUE,
-        f.DESCRIPTION
+			d.DESCRIPTION AS DEVICE_TYPE_NAME,
+			dc.DEVICE_TYPE_CLUSTER_ID,
+      dc.CLUSTER_REF,
+			dc.CLUSTER_NAME,
+			dc.INCLUDE_SERVER,
+			dc.INCLUDE_CLIENT,
+			df.DEVICE_TYPE_CLUSTER_CONFORMANCE,
+			f.FEATURE_ID,
+			f.NAME AS FEATURE_NAME,
+			f.CODE,
+			f.BIT,
+			f.DESCRIPTION,
+			etc.ENDPOINT_TYPE_CLUSTER_ID,
+			eta.ENDPOINT_TYPE_ATTRIBUTE_ID AS FEATUREMAP_ATTRIBUTE_ID,
+			eta.DEFAULT_VALUE AS FEATUREMAP_VALUE
     FROM
-        DEVICE_TYPE d
+			DEVICE_TYPE d
     JOIN
-        DEVICE_TYPE_CLUSTER dc
+			DEVICE_TYPE_CLUSTER dc
     ON 
-        d.DEVICE_TYPE_ID = dc.DEVICE_TYPE_REF
+			d.DEVICE_TYPE_ID = dc.DEVICE_TYPE_REF
     JOIN
-        DEVICE_TYPE_FEATURE df
+			DEVICE_TYPE_FEATURE df
     ON 
-        dc.DEVICE_TYPE_CLUSTER_ID = df.DEVICE_TYPE_CLUSTER_REF
+			dc.DEVICE_TYPE_CLUSTER_ID = df.DEVICE_TYPE_CLUSTER_REF
+    JOIN
+			FEATURE f
+    ON
+			df.FEATURE_REF = f.FEATURE_ID
+    JOIN
+			ENDPOINT_TYPE_CLUSTER etc
+    ON
+			dc.CLUSTER_REF = etc.CLUSTER_REF
     JOIN
-        FEATURE f
+			ENDPOINT_TYPE_ATTRIBUTE eta
     ON
-        df.FEATURE_REF = f.FEATURE_ID
+			etc.ENDPOINT_TYPE_CLUSTER_ID = eta.ENDPOINT_TYPE_CLUSTER_REF
+    JOIN
+			ATTRIBUTE a
+    ON
+			eta.ATTRIBUTE_REF = a.ATTRIBUTE_ID
     WHERE
-        d.DEVICE_TYPE_ID IN (${deviceTypeRefsSql})
+			d.DEVICE_TYPE_ID IN (${deviceTypeRefsSql})
+    AND
+			etc.ENDPOINT_TYPE_REF = ?
+    AND
+			a.NAME = 'FeatureMap'
+    AND
+			a.CODE = 65532
+    AND
+			(
+				(dc.INCLUDE_SERVER = 1 AND etc.SIDE = 'server')
+					OR
+				(dc.INCLUDE_CLIENT = 1 AND etc.SIDE = 'client')
+			)
     ORDER BY
-        d.DEVICE_TYPE_ID,
-        dc.CLUSTER_REF,
-        f.FEATURE_ID
+			d.DEVICE_TYPE_ID,
+			dc.CLUSTER_REF,
+			f.FEATURE_ID
     `,
-    deviceTypeRefs
+    arg
   )
   return features.map(dbMapping.map.deviceTypeFeature)
 }
 
+/**
+ * Evaluate the value of a boolean conformance expression that includes terms and operators.
+ * A term can be an attribute, command, event, feature, or conformance abbreviation.
+ * Operators include AND (&), OR (|), and NOT (!).
+ * The '[]' indicates optional conformance if the expression inside true.
+ * Expression containing comma means otherwise conformance. See spec for details.
+ * Examples of conformance expression: 'A & (!B | C)', 'A & B, [!C]'
+ *
+ * @export
+ * @param {*} expression
+ * @param {*} elementMap
+ * @returns 'mandatory', 'optional', 'provisional', or 'notSupported'
+ */
+function evaluateConformanceExpression(expression, elementMap) {
```

Comment:
All these functions are NOT database query functions. The point of `db/query-xxxx.js` is to contain SQL queries.

The functions related to features, but NOT specifically database queries, should be elsewhere.
Move these functions to some other package.